src/enclave/enclave.h

```diff
@@ -45,39 +45,36 @@ namespace enclave
       const raft::Config& raft_config) :
       circuit(enclave_config->circuit),
       writer_factory(circuit, enclave_config->writer_config),
-      rpcsessions(writer_factory),
       n2n_channels(std::make_shared<ccf::NodeToNode>(writer_factory)),
-      node(writer_factory, network, rpcsessions, notifier, timers),
       notifier(writer_factory),
-      cmd_forwarder(
-        std::make_shared<ccf::Forwarder>(rpcsessions, n2n_channels)),
-      rpc_map(std::make_shared<RpcMap>())
+      rpc_map(std::make_shared<RPCMap>()),
+      rpcsessions(std::make_shared<RPCSessions>(writer_factory, rpc_map)),
+      node(writer_factory, network, rpcsessions, notifier, timers),
+      cmd_forwarder(std::make_shared<ccf::Forwarder<ccf::NodeToNode>>(
+        rpcsessions, n2n_channels, rpc_map))
     {
       logger::config::msg() = AdminMessage::log_msg;
       logger::config::writer() = writer_factory.create_writer_to_outside();
 
       REGISTER_FRONTEND(
         rpc_map,
         members,
-        std::make_unique<ccf::MemberCallRpcFrontend>(network, node));
+        std::make_unique<ccf::MemberRpcFrontend>(network, node));
 
       REGISTER_FRONTEND(
         rpc_map, users, ccfapp::get_rpc_handler(network, notifier));
 
       REGISTER_FRONTEND(
         rpc_map, nodes, std::make_unique<ccf::NodeRpcFrontend>(network, node));
 
-      for (auto& r : rpc_map->get_map())
+      for (auto& [actor, fe] : rpc_map->get_map())
       {
-        auto frontend = dynamic_cast<ccf::RpcFrontend*>(r.second.get());
-        frontend->set_sig_intervals(
+        fe->set_sig_intervals(
           signature_intervals.sig_max_tx, signature_intervals.sig_max_ms);
-        frontend->set_cmd_forwarder(cmd_forwarder);
+        fe->set_cmd_forwarder(cmd_forwarder);
       }
 
-      node.initialize(raft_config, n2n_channels, rpc_map);
-      rpcsessions.initialize(rpc_map);
-      cmd_forwarder->initialize(rpc_map);
+      node.initialize(raft_config, n2n_channels, rpc_map, cmd_forwarder);
```

Comment:
Some re-ordering here so we don't have to call `initialize()` on `rpcsessions` and `cmd_forwarder`. This was required to get access to the remove `const ccf::NodeId from` from the `RPCContext`.